implementations/rust/ockam/ockam_api/src/authenticator/credential_issuer/credential_issuer_worker.rs

```diff
@@ -0,0 +1,81 @@
+use core::time::Duration;
+use minicbor::Decoder;
+use tracing::trace;
+
+use crate::authenticator::credential_issuer::CredentialIssuer;
+use crate::authenticator::{secure_channel_required, AuthorityMembersRepository};
+use ockam::identity::{Credentials, Identifier, IdentitySecureChannelLocalInfo};
+use ockam_core::api::{Method, RequestHeader, Response};
+use ockam_core::compat::boxed::Box;
+use ockam_core::compat::sync::Arc;
+use ockam_core::compat::vec::Vec;
+use ockam_core::{Result, Routed, Worker};
+use ockam_node::Context;
+
+/// This struct runs as a Worker to issue credentials based on a request/response protocol
+pub struct CredentialIssuerWorker {
+    credential_issuer: CredentialIssuer,
+}
+
+impl CredentialIssuerWorker {
+    /// Create a new credentials issuer
+    pub fn new(
+        members: Arc<dyn AuthorityMembersRepository>,
+        credentials: Arc<Credentials>,
+        issuer: &Identifier,
+        project_identifier: Option<String>, // Legacy value, should be removed when all clients are updated to the latest version
+        credential_ttl: Option<Duration>,
+    ) -> Self {
+        Self {
+            credential_issuer: CredentialIssuer::new(
+                members,
+                credentials,
+                issuer,
+                project_identifier,
+                credential_ttl,
+            ),
+        }
+    }
+}
+
+#[ockam_core::worker]
+impl Worker for CredentialIssuerWorker {
+    type Context = Context;
+    type Message = Vec<u8>;
```

Comment:
I wrote earlier but we should stop introducing workers with untyped messages (unless I am missing a reason why this can not be the case for the RPC protocol?)

Reply:
I didn't introduce it, I moved it. But I think the main problem here is that we use CBOR for RPC but BARE for the messages. BTW, we could totally switch messages inside a node to use CBOR without breaking any backwards compatibility